NOnion/Network/TorCircuit.fs

```diff
@@ -287,222 +305,434 @@ type TorCircuit
-                                                ]
-                                            HandshakeType = HandshakeType.NTor
-                                            HandshakeData =
-                                                state.GenerateClientMaterial()
+                                            TorCircuitNode.CryptoState =
+                                                TorCryptoState.FromKdfResult
+                                                    kdfResult
+                                                    false
+                                            Window =
+                                                TorWindow
+                                                    Constants.DefaultCircuitLevelWindowParams
                                         }
-                                        |> RelayData.RelayExtend2
-
-                                    circuitState <-
-                                        Extending(
-                                            circuitId,
-                                            handshakeState,
-                                            nodes,
-                                            connectionCompletionSource
-                                        )
-
-                                    do!
-                                        self.UnsafeSendRelayCell
-                                            Constants.DefaultStreamId
-                                            handshakeCell
-                                            None
-                                            true
+                                )
 
-                                    return connectionCompletionSource.Task
-                                }
-                    | _ ->
-                        return
-                            invalidOp
-                                "Circuit is not in a state suitable for extending"
-                }
+                            tcs.SetResult circuitId
+                        | _ ->
+                            failwith
+                                "Unexpected circuit state when receiving Extended cell"
+                    | RelayData.RelayEstablishedIntro _ ->
+                        match circuitState with
+                        | RegisteringAsIntroductionPoint
+                            (
+                                circuitId,
+                                nodes,
+                                _privateKey,
+                                _publicKey,
+                                tcs,
+                                _callback
+                            ) ->
+                            circuitState <-
+                                ReadyAsIntroductionPoint(
+                                    circuitId,
+                                    nodes,
+                                    _privateKey,
+                                    _publicKey,
+                                    _callback
```

Comment:
@aarani these 3 elements are being used, so remove the underscore

Reply:
Fixed